### ISSUE TITLE
Instant Search: dot separator between sort options is visible to screen readers

### ISSUE THREAD

**Issue by bluefuton** [Automattic/jetpack#20791]
### Impacted plugin

Jetpack

### Steps to Reproduce

1. Go to a site with Instant Search enabled.
2. On Firefox, choose 'Inspect Accessibility Properties' on the sorting links.

<img width="647" alt="Screen Shot 2021-08-23 at 15 06 13" src="https://user-images.githubusercontent.com/17325/130384604-9e6ab38e-f17e-4ebd-913d-6011c07f9e8b.png">

The sorting links:

<img width="252" alt="Screen Shot 2021-08-23 at 15 07 21" src="https://user-images.githubusercontent.com/17325/130384631-dc6bd961-7733-4d01-9b8e-55f2999f2f11.png">

### A clear and concise description of what you expected to happen.

The dot separator is a visual artefact and should not be visible to screenreaders.

### What actually happened

Dot shows up in text label and "." is not accessible by keyboard.

### Browser

Firefox
